src/sample/View.java

```diff
@@ -99,51 +101,47 @@ public void changed(ObservableValue<? extends Boolean> observable, Boolean oldVa
 
     private void prepTheGame(Text helpText, CheckBox defaults, TextField textInput, Stage startGUI)
     {
+        startGUI.close();
         int gridSize;
-
+            // if defaults are checked then apply the default 5x5 size
             if (defaults.isSelected())
             {
                 textInput.clear();
                 gridSize = 5;
             }
             else {
                 try {
+                    // else provide the input as the size but with validation on failing to parse
+                    // the input to a number (so empty or letters)
                     gridSize = Integer.parseInt(textInput.getText());
-                } catch (NumberFormatException e) {
-                    helpText.setText("Please enter a number or check 'defaults'");
+                    if (gridSize > 15)
+                    {
+                        gridSize = 15;
+                        System.out.println("Applying the max as input exceeded it");
+                    }
+                }
+                catch (NumberFormatException e) {
+                    helpText.setText("Please enter a number <= 15 or check 'defaults'");
                     textInput.clear();
                     return;
                 }
             }
+            // apply the computed grid size to the grid pane used in main which
+            // is the central grid pane resource for setting up the game gui
             Main.gridSizeForGame = gridSize;
-            startGameGUI(gridSize, startGUI, defaults.isSelected(), helpText);
+            // start the actual game gui
+            startGameGUI(gridSize, startGUI);
         }
 
 
-    private void startGameGUI(int parseInt, Stage startGUI, boolean defaultsOn, Text help)
+    private void startGameGUI(int parseInt, Stage startGUI)
         {
-            startGUI.close();
-
-            Stage mainGame = new Stage();
+            // creating the new game stage gui and view
+            // Stage mainGame = new Stage();
             View gameView = new View();
-            if (defaultsOn)
-            {
-                gameView.gridSize = 5;
-                gameView.setGridSize(5);
-            }
-            else
-            {
-                try
-                {
-                    gameView.setGridSize(parseInt);
-                }
-                catch (NumberFormatException numException)
-                {
-                    help.setText("You cannot pass an empty value. Either enter a number or" +
-                            "click the default checkbox.");
+            // setting the view grid size
+            gameView.setGridSize(parseInt);
 
-                }
-            }
             Main.gameView = gameView;
```

Comment:
I don't think this is what we should do, we should probably be returning something instead.  I want to avoid this tight coupling as much as possible.  If we have to debug this at some point, this is a hidden set that we'd really struggle to find.

Reply:
May need a review session on this one then. I am not sure where to go here.